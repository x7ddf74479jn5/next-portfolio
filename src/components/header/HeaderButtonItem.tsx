import styles from "src/styles/components/header/HeaderButton.module.scss";

import type { ButtonType } from "../../types/utils/index";

type Props = ButtonType & {
  label: string;
  icon: {
    path: string;
    alt: string;
  };
};

const HeaderButtonItem: React.VFC<Props> = ({ onClick, icon, label }) => {
  return (
    <button className={styles.buttonItem} onClick={onClick}>
      <img src={icon?.path} alt={icon?.alt} />
      <span>{label}</span>
    </button>
  );
};

export default HeaderButtonItem;
