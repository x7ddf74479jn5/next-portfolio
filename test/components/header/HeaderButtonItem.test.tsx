import userEvent from "@testing-library/user-event";
import HeaderButtonItem from "src/components/header/HeaderButtonItem";
import type { RenderResult } from "test/test-utils";
import { reTestCase } from "test/test-utils";
import { render } from "test/test-utils";

describe("HeaderButtonItem", () => {
  let renderResult: RenderResult;
  let mockHandleClick: jest.Mock<any, any>;
  const icon = {
    path: "img.png",
    alt: "alt",
  };

  beforeEach(() => {
    mockHandleClick = jest.fn();
    renderResult = render(<HeaderButtonItem icon={icon} label="label" onClick={mockHandleClick} />);
  });

  it("matches snapshot", () => {
    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  it("rendered correctly", () => {
    expect(renderResult.getByRole("button")).toBeInTheDocument();
    expect(renderResult.getByText("label")).toBeInTheDocument();
    const imgElement = renderResult.getByRole("img");
    expect(imgElement).toHaveAttribute("alt", reTestCase.anyWord);
    expect(imgElement).toHaveAttribute("src", reTestCase.anyImage);
  });

  it("user Interaction", () => {
    userEvent.click(renderResult.getByRole("button"));
    expect(mockHandleClick).toBeCalled();
  });
});
