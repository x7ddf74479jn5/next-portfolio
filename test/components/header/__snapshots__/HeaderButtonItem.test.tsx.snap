// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderButtonItem matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="buttonItem"
  >
    <img
      alt="alt"
      src="img.png"
    />
    <span>
      label
    </span>
  </button>
</DocumentFragment>
`;
